### PR TITLE
Set base port used in tests based on surefire forkNumber

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -27,6 +27,9 @@
             <include>**/*Test.java</include>
             <include>**/*Tests.java</include>
           </includes>
+          <systemPropertyVariables>
+            <worker.id>$${surefire.forkNumber}</worker.id>
```
Comment (matriv) [inline pom.xml:31]: why is double `$` needed?

File: server/src/testFixtures/java/org/elasticsearch/test/test/ESTestCaseTests.java [hunk 2]
```diff
@@ -190,19 +190,25 @@ public void testRandomValueOtherThan() {
     }
 
     public void testWorkerSystemProperty() {
-        assumeTrue("requires running tests with Gradle", System.getProperty("tests.gradle") != null);
+        assumeTrue(
+            "requires running tests with Gradle",
```
Comment (matriv) [inline server/src/testFixtures/java/org/elasticsearch/test/test/ESTestCaseTests.java:194]:
```suggestion
            "requires running tests with Maven",
```

**Review by matriv — APPROVED** [review]
Left some comments though.

File: pom.xml (hunk 1 above)
Comment (mfussenegger) [reply pom.xml:31]: See https://stackoverflow.com/a/57494619/1333061

Comment (matriv) [reply pom.xml:31]: thx!

**Pull request merged**